src/unused_finder/unused_finder.rs

```diff
@@ -0,0 +1,323 @@
+                    })
+                    .collect();
+                (file.source_file_path.clone(), ids)
+            })
+            .collect();
+        self.flattened_walk_file_data
+            .par_iter_mut()
+            .for_each(|file| {
+                process_import_export_info(file, &resolver);
+            });
+
+        self.refresh_graph();
+    }
+
+    fn refresh_graph(&mut self) {
+        let files: HashMap<String, Arc<GraphFile>> = self
+            .flattened_walk_file_data
+            .par_iter()
+            .map(|file| {
+                (
+                    file.source_file_path.to_string(),
+                    Arc::new(GraphFile::new(
+                        file.source_file_path.clone(),
+                        file.import_export_info
+                            .exported_ids
+                            .iter()
+                            .map(|e| e.metadata.export_kind.clone())
+                            .collect(),
+                        file.import_export_info.clone(),
+                        self.entry_packages.clone().contains(&file.package_name), // mark files from entry_packages as used
+                    )),
+                )
+            })
+            .collect();
+
+        self.graph = Graph { files };
+    }
+
+    #[napi]
+    pub fn find_unused_items(
+        &mut self,
+        files_to_check: Vec<String>,
+    ) -> napi::Result<UnusedFinderReport> {
+        self.logs = vec![];
+        self.logs.push(format!("{:?}", &files_to_check));
+        if files_to_check
+            .iter()
+            .any(|file| !self.file_path_exported_items_map.contains_key(file))
+        {
+            self.logs.push("Refreshing all files".to_string());
+            self.refresh_file_list();
+            self.refresh_graph();
+        } else {
+            self.logs.push("Refreshing only some files!".to_string());
+            self.refresh_files_to_check(&files_to_check);
+        }
+        self.file_path_exported_items_map =
+            create_report_map_from_flattened_files(&self.flattened_walk_file_data);
+        let mut file_path_exported_items_map = self.file_path_exported_items_map.clone();
+        let entry_files: Vec<String> = self
+            .flattened_walk_file_data
+            .par_iter()
+            .filter_map(|file| {
+                if self.entry_packages.contains(&file.package_name) {
+                    return Some(file.source_file_path.clone());
+                }
+                None
+            })
+            .collect();
+        let mut frontier = entry_files;
+        for i in 0..10_000_000 {
+            frontier = self.graph.bfs_step(frontier);
+
+            if frontier.is_empty() {
+                break;
+            }
+            if i == 10_000_000 {
+                return Err(napi::Error::new(
+                    napi::Status::GenericFailure,
+                    "exceeded max iterations".to_string(),
+                ));
+            }
```

Comment:
rather than check a conditional on every loop iter, check if frontier.is_empty() after the loop completes. Should run faster, and avoids needing to repeat the 10_000_000 constant